### PR TITLE
Create reusable date input

### PR DESCRIPTION
**Pull request by brittanylau**
### Notion ticket link

[Make reusable form field inputs](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=42b46b9b76d740a29585de1d2fd3ed00)

### Implementation description

- created a `DateInput` that opens a date picker
- installs the @material-ui/pickers library (recommended by the [mui docs](https://material-ui.com/components/pickers/#material-ui-pickers))
- will add unit tests later, trying to get this done to unblock a task next sprint

### Steps to test

1. will be used in an upcoming PR

### What should reviewers focus on?

- code quality
- anything missing?

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by Tayef-Shah — APPROVED** [review]
Just wanna mention I had this error after running yarn install for the first time. 
Screen Shot 2021-06-23 at 10 05 38 PM
I tried to reinstall those specific dependencies again, but it didn't work. However, deleting the `yarn.lock` file and running `yarn install` did the trick. I think I remember us having an issue with the package lock file in the past.
One last thing, maybe let's add the date picker to the DOB field in the registration form?

File: src/components/common/inputs/DateInput.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <InputLabel htmlFor={id}>{label}</InputLabel>
+    </Box>
+    <Box width={inputWidth}>
+      <DatePicker
+        id={id}
+        autoOk
+        disableToolbar
+        variant="inline"
+        inputVariant="outlined"
+        format="MMMM D, yyyy"
```
[the remark this answers is not in the document]
Comment (brittanylau) [reply src/components/common/inputs/DateInput.tsx:24]: this is following the [moment format strings](https://momentjs.com/docs/#/parsing/string-format/), which is the date management provider we're using

**Comment by brittanylau** [comment]
> Just wanna mention I had this error after running yarn install for the first time.
> I tried to reinstall those specific dependencies again, but it didn't work. However, deleting the `yarn.lock` file and running `yarn install` did the trick. I think I remember us having an issue with the package lock file in the past.

@Tayef-Shah hmm this is strange, we shouldn't have a package-lock.json file so if you had one in your dev env that might cause some complications 😖 

> One last thing, maybe let's add the date picker to the DOB field in the registration form?

yup this is being done next sprint :) ticket [here](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=b739b89c29204d25a7a8e7f06af111f3)

it should also be used in the edit sidebar form, but I imagine it'll reuse the same component to determine which input to use

File: src/components/common/inputs/DateInput.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tayef-Shah) [reply src/components/common/inputs/DateInput.tsx:24]: ok looks good then 👍 thank you

**Comment by Tayef-Shah** [comment]
@brittanylau oops didn't mean exactly `package-lock.json` I just meant like a dependency lock file in general. I've never installed via npm so I don't have a `package-lock.json` file (phew).

Also thanks for mentioning some details on how we're using the date picker for next sprint.

**Comment by brittanylau** [comment]
@Tayef-Shah ooh I see - I tried reproducing just now and didn't have luck 😞 since the deploy preview also looks ok I'll merge! let me know if you have any other dep issues 🤞

**Pull request merged**